### PR TITLE
Added an option to pause and autosave on desync

### PR DESCRIPTION
**Pull request by SokyranTheDragon**

### CODE REVIEW

**Review by Nebual — APPROVED** [review]
Nice, should make reconnecting speedier.

**Review by notfood — APPROVED** [review]
I like it. Very clean.

File: Languages/English/Keyed/Multiplayer.xml [hunk 1]
```diff
@@ -146,5 +146,8 @@
   <MpAggressiveTicking>Aggressive ticking</MpAggressiveTicking>
   <MpAggressiveTickingDesc>Always try to automatically catch up with the server at the cost of FPS.</MpAggressiveTickingDesc>
   <MpMultiplayerCompatibility>MP Compatibility</MpMultiplayerCompatibility>
+
+  <MpAutosaveOnDesync>Pause and autosave on desync</MpAutosaveOnDesync>
+  <MpAutosaveOnDesyncDesc>If any of the players desyncs the game will be paused and saved automatically. This should make it easier to rejoin without having to simulate the game, and could be useful if the host cannot save the game themselves.</MpAutosaveOnDesyncDesc>
```
Comment (Cameron-Friel) [inline Languages/English/Keyed/Multiplayer.xml:151]: Sorry to nitpick, but I think this should be written as "If any of the players desync, the game will be paused and saved automatically.". Other than that I believe this is a neat quality of life update!

**Comment by notfood** [comment]
Unfortunately, this patch causes that the disconnected player gets stuck in pause on reconnect for some reason.

**Comment by SokyranTheDragon** [comment]
This should fix the issue (and it did on my end).

**Comment by notfood** [comment]
Seems to work, but I've been getting duplicate IDs even though I never made the save single player. I'm trying to figure out if it generates here.

**Comment by notfood** [comment]
Works fine

**Pull request merged**